external_services/lti.py

```diff
@@ -60,9 +62,9 @@ def __init__(self, service, user, instance, request, title, context_id=None, lin
             "launch_presentation_locale": get_language(),
             "launch_presentation_document_target":
                 "iframe" if exercise and exercise.open_in_iframe else "window",
-            "launch_presentation_return_url": request.scheme + '://' + request.get_host() + instance.get_absolute_url(),
+            "launch_presentation_return_url": settings.BASE_URL + instance.get_absolute_url(),
 
-            "tool_consumer_instance_guid": request.get_host() + "/aplus",
+            "tool_consumer_instance_guid": parsed_url.hostname + ":"+ str(parsed_url.port) + "/aplus",
```

Comment:
This `parsed_url.hostname + ":"+ str(parsed_url.port)` (or alternative, if changed) should be calculated only once in the function.